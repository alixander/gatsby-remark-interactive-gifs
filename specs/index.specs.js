const remark = require("remark")
const plugin = require("../src/index")

/*
 * dolphin.gif courtesy of giphy.com:
 * > https://giphy.com/gifs/dolphin-Vt4wDrLAnH1tu
 *
 * play and placeholder images courtesy of flaticon:
 * > https://www.flaticon.com/free-icon/play_2819037
 * > https://www.flaticon.com/free-icon/photo_2793644
 */

const getNodeContent = node => node.children[0].children[0]

const options = {
  pwd: `${__dirname}`,
  relativePath: `/static/gifs`,
  src: `${__dirname}/images`,
  dest: `${__dirname}/images`,
  play: `${__dirname}/images/play.gif`,
  placeholder: `${__dirname}/images/placeholder.gif`,
}

describe(`gatsby-remark-interactive-gifs`, () => {

  it(`inline code not matching the protocol`, async () => {
    const markdown = "`npm i gatsby-remark-interactive-gifs`"
    const markdownAST = remark().parse(markdown)
    const processed = await plugin({ markdownAST }, options)
    expect(processed).toBeTruthy()
    expect(getNodeContent(markdownAST)).toMatchSnapshot()
  })

  it(`should embed the placeholder when the image does not exist`, async () => {
    const markdown = "`gif:porpoise.gif`"
    const markdownAST = remark().parse(markdown)
    const processed = await plugin({ markdownAST }, options)
    expect(processed).toBeTruthy()
    expect(getNodeContent(markdownAST)).toMatchSnapshot()
  })

  it(`should embed the gif`, async () => {
    const markdown = "`gif:dolphin.gif`"
    const markdownAST = remark().parse(markdown)
    const processed = await plugin({ markdownAST }, options)
    expect(processed).toBeTruthy()
    expect(getNodeContent(markdownAST)).toMatchSnapshot()
  })

  it(`should embed the gif with a specified element id`, async () => {
    const markdown = "`gif:dolphin.gif:id=i-am-a-dolphin`"
    const markdownAST = remark().parse(markdown)
    const processed = await plugin({ markdownAST }, options)
    expect(processed).toBeTruthy()
    expect(getNodeContent(markdownAST)).toMatchSnapshot()
  })

  it(`should embed the gif with a caption`, async () => {
    const markdown = "`gif:dolphin.gif:caption=So long and thanks for all the fish`"
    const markdownAST = remark().parse(markdown)
    const processed = await plugin({ markdownAST }, options)
    expect(processed).toBeTruthy()
    expect(getNodeContent(markdownAST)).toMatchSnapshot()
  })

  it(`should embed the gif with a class`, async () => {
    const markdown = "`gif:dolphin.gif:class=do-not-stretch-me`"
    const markdownAST = remark().parse(markdown)
    const processed = await plugin({ markdownAST }, options)
    expect(processed).toBeTruthy()
    expect(getNodeContent(markdownAST)).toMatchSnapshot()
  })

  it(`should embed the gif with an id, caption and class`, async () => {
    const markdown = "`gif:dolphin.gif:caption=So long and thanks for all the fish;class=do-not-stretch-me;id=am-i-a-fishy`"
    const markdownAST = remark().parse(markdown)
    const processed = await plugin({ markdownAST }, options)
    expect(processed).toBeTruthy()
    expect(getNodeContent(markdownAST)).toMatchSnapshot()
  })
})
