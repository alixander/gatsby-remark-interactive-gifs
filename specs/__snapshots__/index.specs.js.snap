// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gatsby-remark-interactive-gifs inline code not matching the protocol 1`] = `
Object {
  "position": Position {
    "end": Object {
      "column": 39,
      "line": 1,
      "offset": 38,
    },
    "indent": Array [],
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "inlineCode",
  "value": "npm i gatsby-remark-interactive-gifs",
}
`;

exports[`gatsby-remark-interactive-gifs should embed the gif 1`] = `
Object {
  "position": Position {
    "end": Object {
      "column": 18,
      "line": 1,
      "offset": 17,
    },
    "indent": Array [],
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "html",
  "value": "
      <div class=\\"interactive-gif \\">
        <div class=\\"embedded\\" style=\\"padding-top: 83.6%\\">
          <div id=\\"loading-dolphin.gif\\"
              class=\\"loading\\" style=\\"background-size: cover; background-image: url('/static/gifs/still-dolphin.gif');\\">
              <img class=\\"indicator\\" src=\\"/static/gifs/loading.gif\\" />
          </div>
          <div id=\\"dolphin.gif\\"
              class=\\"gif-container\\"
              style=\\"display: none;\\"
              onclick=\\"document.getElementById('dolphin.gif').style.display = 'none';
                        document.getElementById('still-dolphin.gif').style.display = 'block';\\">
            <img id=\\"image-dolphin.gif\\"
                class=\\"gif\\"
                data-original=\\"/static/gifs/dolphin.gif\\" />
          </div>

          <div id=\\"still-dolphin.gif\\"
              class=\\"still-container\\"
              onclick=\\"var gif = document.getElementById('image-dolphin.gif');
                        gif.src = gif.dataset.original + '?t=' + new Date().getTime();
                        document.getElementById('still-dolphin.gif').style.display = 'none';
                        document.getElementById('dolphin.gif').style.display = 'block';\\">
            <img id=\\"image-still-dolphin.gif\\"
                class=\\"still\\"
                src=\\"/static/gifs/still-dolphin.gif\\" />
            <img class=\\"play\\"
                src=\\"/static/gifs/play.gif\\" />
          </div>
        </div>
        <div class=\\"caption\\"></div>
      </div>
    ",
}
`;

exports[`gatsby-remark-interactive-gifs should embed the gif with a caption 1`] = `
Object {
  "position": Position {
    "end": Object {
      "column": 62,
      "line": 1,
      "offset": 61,
    },
    "indent": Array [],
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "html",
  "value": "
      <div class=\\"interactive-gif \\">
        <div class=\\"embedded\\" style=\\"padding-top: 83.6%\\">
          <div id=\\"loading-dolphin.gif\\"
              class=\\"loading\\" style=\\"background-size: cover; background-image: url('/static/gifs/still-dolphin.gif');\\">
              <img class=\\"indicator\\" src=\\"/static/gifs/loading.gif\\" />
          </div>
          <div id=\\"dolphin.gif\\"
              class=\\"gif-container\\"
              style=\\"display: none;\\"
              onclick=\\"document.getElementById('dolphin.gif').style.display = 'none';
                        document.getElementById('still-dolphin.gif').style.display = 'block';\\">
            <img id=\\"image-dolphin.gif\\"
                class=\\"gif\\"
                data-original=\\"/static/gifs/dolphin.gif\\" />
          </div>

          <div id=\\"still-dolphin.gif\\"
              class=\\"still-container\\"
              onclick=\\"var gif = document.getElementById('image-dolphin.gif');
                        gif.src = gif.dataset.original + '?t=' + new Date().getTime();
                        document.getElementById('still-dolphin.gif').style.display = 'none';
                        document.getElementById('dolphin.gif').style.display = 'block';\\">
            <img id=\\"image-still-dolphin.gif\\"
                class=\\"still\\"
                src=\\"/static/gifs/still-dolphin.gif\\" />
            <img class=\\"play\\"
                src=\\"/static/gifs/play.gif\\" />
          </div>
        </div>
        <div class=\\"caption\\">So long and thanks for all the fish</div>
      </div>
    ",
}
`;

exports[`gatsby-remark-interactive-gifs should embed the gif with a class 1`] = `
Object {
  "position": Position {
    "end": Object {
      "column": 42,
      "line": 1,
      "offset": 41,
    },
    "indent": Array [],
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "html",
  "value": "
      <div class=\\"interactive-gif do-not-stretch-me\\">
        <div class=\\"embedded\\" style=\\"padding-top: 83.6%\\">
          <div id=\\"loading-dolphin.gif\\"
              class=\\"loading\\" style=\\"background-size: cover; background-image: url('/static/gifs/still-dolphin.gif');\\">
              <img class=\\"indicator\\" src=\\"/static/gifs/loading.gif\\" />
          </div>
          <div id=\\"dolphin.gif\\"
              class=\\"gif-container\\"
              style=\\"display: none;\\"
              onclick=\\"document.getElementById('dolphin.gif').style.display = 'none';
                        document.getElementById('still-dolphin.gif').style.display = 'block';\\">
            <img id=\\"image-dolphin.gif\\"
                class=\\"gif\\"
                data-original=\\"/static/gifs/dolphin.gif\\" />
          </div>

          <div id=\\"still-dolphin.gif\\"
              class=\\"still-container\\"
              onclick=\\"var gif = document.getElementById('image-dolphin.gif');
                        gif.src = gif.dataset.original + '?t=' + new Date().getTime();
                        document.getElementById('still-dolphin.gif').style.display = 'none';
                        document.getElementById('dolphin.gif').style.display = 'block';\\">
            <img id=\\"image-still-dolphin.gif\\"
                class=\\"still\\"
                src=\\"/static/gifs/still-dolphin.gif\\" />
            <img class=\\"play\\"
                src=\\"/static/gifs/play.gif\\" />
          </div>
        </div>
        <div class=\\"caption\\"></div>
      </div>
    ",
}
`;

exports[`gatsby-remark-interactive-gifs should embed the gif with a specified element id 1`] = `
Object {
  "position": Position {
    "end": Object {
      "column": 36,
      "line": 1,
      "offset": 35,
    },
    "indent": Array [],
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "html",
  "value": "
      <div class=\\"interactive-gif \\">
        <div class=\\"embedded\\" style=\\"padding-top: 83.6%\\">
          <div id=\\"loading-i-am-a-dolphin\\"
              class=\\"loading\\" style=\\"background-size: cover; background-image: url('/static/gifs/still-dolphin.gif');\\">
              <img class=\\"indicator\\" src=\\"/static/gifs/loading.gif\\" />
          </div>
          <div id=\\"i-am-a-dolphin\\"
              class=\\"gif-container\\"
              style=\\"display: none;\\"
              onclick=\\"document.getElementById('i-am-a-dolphin').style.display = 'none';
                        document.getElementById('still-i-am-a-dolphin').style.display = 'block';\\">
            <img id=\\"image-i-am-a-dolphin\\"
                class=\\"gif\\"
                data-original=\\"/static/gifs/dolphin.gif\\" />
          </div>

          <div id=\\"still-i-am-a-dolphin\\"
              class=\\"still-container\\"
              onclick=\\"var gif = document.getElementById('image-i-am-a-dolphin');
                        gif.src = gif.dataset.original + '?t=' + new Date().getTime();
                        document.getElementById('still-i-am-a-dolphin').style.display = 'none';
                        document.getElementById('i-am-a-dolphin').style.display = 'block';\\">
            <img id=\\"image-still-i-am-a-dolphin\\"
                class=\\"still\\"
                src=\\"/static/gifs/still-dolphin.gif\\" />
            <img class=\\"play\\"
                src=\\"/static/gifs/play.gif\\" />
          </div>
        </div>
        <div class=\\"caption\\"></div>
      </div>
    ",
}
`;

exports[`gatsby-remark-interactive-gifs should embed the gif with an id, caption and class 1`] = `
Object {
  "position": Position {
    "end": Object {
      "column": 102,
      "line": 1,
      "offset": 101,
    },
    "indent": Array [],
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "html",
  "value": "
      <div class=\\"interactive-gif do-not-stretch-me\\">
        <div class=\\"embedded\\" style=\\"padding-top: 83.6%\\">
          <div id=\\"loading-am-i-a-fishy\\"
              class=\\"loading\\" style=\\"background-size: cover; background-image: url('/static/gifs/still-dolphin.gif');\\">
              <img class=\\"indicator\\" src=\\"/static/gifs/loading.gif\\" />
          </div>
          <div id=\\"am-i-a-fishy\\"
              class=\\"gif-container\\"
              style=\\"display: none;\\"
              onclick=\\"document.getElementById('am-i-a-fishy').style.display = 'none';
                        document.getElementById('still-am-i-a-fishy').style.display = 'block';\\">
            <img id=\\"image-am-i-a-fishy\\"
                class=\\"gif\\"
                data-original=\\"/static/gifs/dolphin.gif\\" />
          </div>

          <div id=\\"still-am-i-a-fishy\\"
              class=\\"still-container\\"
              onclick=\\"var gif = document.getElementById('image-am-i-a-fishy');
                        gif.src = gif.dataset.original + '?t=' + new Date().getTime();
                        document.getElementById('still-am-i-a-fishy').style.display = 'none';
                        document.getElementById('am-i-a-fishy').style.display = 'block';\\">
            <img id=\\"image-still-am-i-a-fishy\\"
                class=\\"still\\"
                src=\\"/static/gifs/still-dolphin.gif\\" />
            <img class=\\"play\\"
                src=\\"/static/gifs/play.gif\\" />
          </div>
        </div>
        <div class=\\"caption\\">So long and thanks for all the fish</div>
      </div>
    ",
}
`;

exports[`gatsby-remark-interactive-gifs should embed the placeholder when the image does not exist 1`] = `
Object {
  "position": Position {
    "end": Object {
      "column": 19,
      "line": 1,
      "offset": 18,
    },
    "indent": Array [],
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "html",
  "value": "
      <div class=\\"interactive-gif\\">
        <div class=\\"placeholder\\">
          <img src=\\"/static/gifs/placeholder.gif\\" />
        </div>
      </div>
    ",
}
`;
